File: Data Structures & Algorithms/palindromic-substrings/submission-0.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int count =  0;

        for(int i = 0;i < s.length();i++){
            for(int j = 0;j <= 1;j++){
                int left = i;
                int right = i + j;

                while(left >= 0  && right < s.length() && s[left] == s[right]){
                    count++;
                    left--;
                    right++;
                }
            }
        }

        return count;
    }
};

File: Data Structures & Algorithms/serialize-and-deserialize-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Codec:
    
    # Encodes a tree to a single string.
    def serialize(self, root: Optional[TreeNode]) -> str:
        if root == None:
            return "N"
        res = []
        q = deque([root])
        while len(q) != 0:
            temp = q[0]
            q.popleft()

            if(temp == None):
                res.append("N")
            else:
                res.append(str(temp.val))
                q.append(temp.left)
                q.append(temp.right)
            
        return ",".join(res)
    # Decodes your encoded data to tree.
    def deserialize(self, data: str) -> Optional[TreeNode]:
        vals = data.split(",")
        if vals[0] == "N":
            return None
        root = TreeNode(int(vals[0]))
        q = deque([root])
        index = 1
        
        while q:
            node = q.popleft()
            if vals[index] != "N":
                node.left = TreeNode(int(vals[index]))
                q.append(node.left)
            index += 1
            
            if vals[index] != "N":
                node.right = TreeNode(int(vals[index]))
                q.append(node.right)
            index += 1
        return root